openedx_ledger/models.py

```diff
@@ -617,3 +641,99 @@ class Adjustment(TimeStampedModelWithUuid):
         ),
     )
     history = HistoricalRecords()
+
+
+class SalesContractReferenceProvider(TimeStampedModel):
+    """
+    All possible providers for sales contracts.
+
+    Typically your sales contracts are recorded within one centralized system-of-record (e.g. SalesForce), in which case
+    this table may only ever have one single record to describe that provider.
+
+    .. no_pii:
+    """
+
+    class Meta:
+        """
+        Metaclass for SalesContractReferenceProvider.
+        """
+
+        verbose_name = 'Sales Contract Reference Provider'
+
+    name = models.CharField(
+        max_length=255,
+        help_text="The pretty name of the contract reference provider.",
+    )
+    slug = models.SlugField(
+        max_length=32,
+        unique=True,
+        help_text="A unique slug for this sales contract reference provider.",
+    )
+
+    def __str__(self):
+        """
+        Return string representation of this external fulfillment provider, visible in logs, django admin, etc.
+        """
+        msg = (
+            f'<SalesContractReferenceProvider\n'
+            f'name={self.name}\n'
+            f'slug={self.slug}>'
+        )
+        return msg
+
+
+class Deposit(TimeStampedModelWithUuid):
+    """
+    Represents a deposit into to the balance of a ledger via a transaction (with quantity > 0).
+
+    .. no_pii:
+    """
+    ledger = models.ForeignKey(
+        Ledger,
+        related_name='deposits',
+        null=False,
+        on_delete=models.CASCADE,
+        help_text=(
+            "The Ledger instance with which this deposit is associated."
+        )
+    )
+    # This field is non-contributing, and should NOT be factored into balance calculations. It should only be used to
+    # accept input from an admin form and feed the transaction creation.
+    desired_deposit_quantity = models.BigIntegerField(
```

Comment:
I forget how this is on the other models... Should we be naming things like this `desired_deposit_quantity_cents`?

Reply:
We've been avoiding hard-coding units across this entire openedx-ledger library. The implied unit is whatever unit the Ledger declares. https://github.com/openedx/openedx-ledger/blob/cd7a60c6d9f829018e7ec84e00c208dbc20c00dc/openedx_ledger/models.py#L129